Segy

aaronjgirard opened the pull request:
## Description

Add support to read SEGY files into a single patch in a spool.

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [x] documented the new feature with docstrings or appropriate doc page.
- [x] included a test. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [x] your name has been added to the contributors page (docs/contributors.md).
- [x] added the "ready_for_review" tag once the PR is ready to be reviewed.

d-chambers commented:
Hey @aaronjgirard,

A few small changes:

1. One test was failing locally because `file_path` should be named `path` in the `PatchAttrs` construct. My bad, I mislead you here.

2. There were a few RSF files left. I have deleted them as they should go in a different PR (and were causing codecov to complain). 

I was thinking more about partial reads, and I think it wont be too hard to do. Will check on it next week. I will also test against the Stanford segy files on PubDAS and if all that works it should be good to merge. Thanks for working on this.

aaronjgirard commented:
@d-chambers Thanks for fixing those and I hope it works with the Stanford files in PubDAS.

 I fixed the docstring and added my name to the contributors list on docs/contributors.md

Please let me know if I need to do anything else or if I skipped any steps.

d-chambers commented:
Quick update: 

I just tested locally on my machine and this works fine for all our pubDAS sample files. I also pushed a small change to support partial reads.
